components/style/properties/data.py

```diff
@@ -90,6 +90,8 @@ def __init__(self, style_struct, name, spec="NO SPEC", animatable=None, derived_
                  allowed_in_keyframe_block=True, complex_color=False, cast_type='u8',
                  has_uncacheable_values=False, logical=False):
         self.name = name
+        if not spec:
+            raise TypeError("Spec should be specified for %s" %name)
```

Comment:
nit: I think the common style is with spaces around both sides of `%`? Here and below.